### PR TITLE
Creating Cook Recipe button and Clean Up code for Final Product w/o POST

### LINKED ISSUE #14: User Favorite Recipes DOM TDD
As a user, I want to be able to see all of my favorite recipes at a glance for each recipe or sort them to only show my favorite recipes.

- [ ] Create an image that is on each recipe card that can toggle state between favorite or unfavorite.
- [ ] Have a link or a tag that will sort a user's favorite recipes while hiding all of the other recipes.
- [ ] Happy and sad path testing if applicable.

### LINKED ISSUE #16: User Recipe to Cook Doom TDD
As a user, I want to be able to see on my recipe card a recipe that I want to cook for this week.

- [ ] Create a place on the recipe card that will show a user (graphic or tag?) that they have selected this recipe card to cook for the week.
- [ ] Happy and sad path if applicable.

### LINKED ISSUE #29: Pantry Ingredients In Pantry DOM
As a user, I want to be able to see how many ingredients I have inside of my pantry when I have selected a specific meal to cook.

- [x] Create a page (modal?) when a user has clicked on a recipe that shows them the specific of the recipe and when they have enough ingredients to cook it.

### LINKED ISSUE #48: UX Issue: modal closing button
modal for recipe: [x] button on R instead of L-hand size

### LINKED ISSUE #73: Discuss what our MVP is

### LINKED ISSUE #79: Determine if All Functions ES6
- [ ] Check in with instructors about whether or not all functions should be refactored to ES6

### LINKED ISSUE #88: Refactor: addToMyRecipes(event)
This functionality should live somewhere else (opening and closing recipe details)
```
  } else if (event.target.id === "exit-recipe-btn") {
    exitRecipe();
  } else if (isDescendant(event.target.closest(".recipe-card"), event.target)) {
    openRecipeInfo(event);
  }
```

### PR DESCRIPTION
**Pull request by JWoltjen**
### What’s this PR do?  

This is the final substantive PR in the project. It creates a Cook Recipe button for every recipe card, it comments away the unused vestigial code and the pseudocode. 

### Where should the reviewer start?  

The user should start in scripts.js, move on to fetch.js, and verify that pantry.js and user.js are in working order. 

### How should this be manually tested?  

The product should be manually tested (and has been manually tested) by firing up the server and manipulating the DOM to ensure that all functions are working properly. 

### What are the relevant tickets?  
closes: #79 
closes: #88 
closes: #73 
closes: #48 
closes: #29 
closes: #16 
closes: #14

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -197,20 +123,31 @@ function addToMyRecipes(event) {
...
-      console.log(user.favoriteRecipes)
     } else {
       event.target.src = "../images/apple-logo-outline.png";
       event.target.setAttribute('aria-pressed', false);
       user.removeRecipe(cardId);
-      console.log(user.favoriteRecipes)
       showSavedRecipes()
     }
-  } else if (event.target.id === "exit-recipe-btn") {
+  }else if (event.target.id === "cook-recipe-btn"){
```
Comment (aemcdonald) [inline src/scripts.js:132]: Great call using event bubbling here

File: src/domUpdates.js [hunk 2]
```diff
@@ -79,21 +79,23 @@ const domUpdates = {
...
+        <button id=cook-recipe-btn>Cook recipe</button>
       </section>`
+
     main.insertAdjacentHTML("beforeend", cardHtml);
   },
 
   openRecipeInfo(event) {
     let fullRecipeInfo = document.querySelector(".recipe-instructions");
     fullRecipeInfo.style.display = "inline";
-    let recipeId = event.path.find(e => e.id).id;
```
Comment (npdarrington) [inline src/domUpdates.js:89]: Thank you for implementing this new change. I'm not entirely sure why event.path was working at the beginning of the project and broke (specifically on firefox) towards the end of the project. Let's do some research on this for future learning.

File: src/domUpdates.js [hunk 3]
```diff
@@ -79,21 +79,23 @@ const domUpdates = {
...
+    let recipeId = event.target.closest('.recipe-card').id
```
Comment (hannakim91) [inline src/domUpdates.js:91]: Good call as event.path is not yet fully supported as we discussed in the group!

**Review by npdarrington — APPROVED** [review]

File: src/scripts.js [hunk 4]
```diff
@@ -197,20 +123,31 @@ function addToMyRecipes(event) {
...
+    cookRecipe(event);
+  }else if (event.target.id === "exit-recipe-btn") {
     domUpdates.exitRecipe();
   } else if (isDescendant(event.target.closest(".recipe-card"), event.target)) {
     domUpdates.openRecipeInfo(event);
   }
 }
+function cookRecipe(event) {
```
Comment (aemcdonald) [inline src/scripts.js:140]: Great job with this function here & iterating over recipe data to make sure the recipe matches. I wish we had more time to get our POSTs working!

File: src/scripts.js [hunk 5]
```diff
@@ -40,8 +37,7 @@ let ingredientsData;
 let recipeData;
 let recipe;
 let recipes = []
-let pantry;
-let users; //do we need this still?
+let users;
```
Comment (hannakim91) [inline src/scripts.js:40]: Looks like we were able to clean up some unused global variables by making use of the domUpdates object

**Review by aemcdonald — APPROVED** [review]
Great job Jeff! I wish we had more time to get our POSTs to work but I think it is a good call to not begin refactoring methods so close to our submission time.

**Review by hannakim91 — APPROVED** [review]
I agree it makes sense to submit the project without POST feature working - we would've had to do a lot of refactoring to get the code to match how the spec wanted us to organize the fetch body

**Pull request merged**